Automattic/jetpack
# Limit post_content that gets Synced to 5MB

Posts can have content of any length. We've seen that some sites send post_content well exceeding 10MB at a time which leads to processing errors and Sync no longer functional on their site as the post continually tries to be synced and processed.

This PR will truncate post_content that exceeds 5MB so that we can still sync the overall object but the post_content will be empty.

#### Changes proposed in this Pull Request:
* Limit post_content that is syncable to 5MB.


#### Does this pull request change what data or activity we track or use?
* No

#### Testing instructions:
Unit Tests added to test the imposed limit.

Manual Testing
* create a post with post_content exceeding 5000000 characters in length. (str_pad can be useful)
* $post_sync_module = Modules::get_module( 'posts' );
* $filtered_post = $post_sync_module->filter_post_content_and_add_links( $post );
* Verify $filtered_post->post_content is empty.
* Create a post with post_content less than 5000000 characters in length
* $post_sync_module = Modules::get_module( 'posts' );
* $filtered_post = $post_sync_module->filter_post_content_and_add_links( $post );
* Verify $filtered_post->post_content was not modified.

#### Proposed changelog entry for your changes:
* Jetpack_Sync :: Impose a max limit of 5MB on post_content that is synced.
